### PR TITLE
feat(wallet-connect): update safe-apps-sdk to enable synchronous off-chain signatures

### PR DESCRIPTION
**Pull request by schmanu**
Depends on https://github.com/safe-global/safe-apps-sdk/pull/458
UI implementation: https://github.com/safe-global/web-core/pull/1630

### CODE REVIEW

File: apps/wallet-connect/src/hooks/useWalletConnectV1.tsx [hunk 1]
```diff
@@ -62,9 +62,13 @@ const useWalletConnectV1 = (
         trackEvent(NEW_SESSION_ACTION, WALLET_CONNECT_VERSION_1, wcConnector.peerMeta)
 
         setWcClientData(payload.params[0].peerMeta)
+
+        // HACK: Always enable off-chain msgs
+        web3Provider.send('safe_setSettings', [{ offChainSigning: true }])
```
Comment (katspaugh) [inline apps/wallet-connect/src/hooks/useWalletConnectV1.tsx:67]: @iamacook how can we make this work for OpenSea w/o this hack?
Basically, not just OpenSea, but any app that supports EIP-1271 will work out of the box with 1/X Safe.

Comment (iamacook) [reply apps/wallet-connect/src/hooks/useWalletConnectV1.tsx:67]: For future reference: after discussing this, we will look into defaulting OpenSea to off-chain for 1/n Safes.

File: apps/wallet-connect/src/hooks/useWalletConnectV2.tsx [hunk 2]
```diff
@@ -168,8 +168,9 @@ const useWalletConnectV2 = (
       // events
       web3wallet.on('session_proposal', async proposal => {
         const { id, params } = proposal
-        const { requiredNamespaces } = params
+        const { requiredNamespaces, optionalNamespaces } = params
```
Comment (schmanu) [inline apps/wallet-connect/src/hooks/useWalletConnectV2.tsx:171]: We could remove this from this release as it's slightly out of scope.

But I realized that some dapps won't work without also taking the `optionalNamespaces` into account.

File: apps/wallet-connect/src/hooks/useWalletConnectV2.tsx [hunk 3]
```diff
@@ -217,7 +218,7 @@ const useWalletConnectV2 = (
                 accounts: safeAccount || [
                   `${EVMBasedNamespaces}:${safe.chainId}:${safe.safeAddress}`,
                 ],
-                methods: EIP155Namespace.methods,
+                methods: [...EIP155Namespace.methods, ...(optionalEIP155Namespace?.methods || [])],
```
Comment (iamacook) [inline apps/wallet-connect/src/hooks/useWalletConnectV2.tsx:221]: If you use `concat`, you need not use a fallback (and it's also more performant).
```suggestion
                methods: EIP155Namespace.methods.concat(optionalEIP155Namespace?.methods),
```

**Review by iamacook — APPROVED** [review]

**Review by mmv08 — APPROVED** [review]
top content

**Pull request merged**